Sources/Localization/MessageFormatter.php

```diff
@@ -865,7 +865,7 @@ protected static function applyNumberSkeleton(int|float|string $number, string $
 
 		// Ensure $number is a string.
 		if (is_float($number)) {
-			$precision = max(0, strlen(rtrim(sprintf('%.53F', abs($number + 0)), '0')) - strlen(strval(intval($number + 0))) - 1);
+			$precision = (int) strpos(strrev(strval($number)), '.') - strlen(strval(intval($number + 0))) - 1;
```

Comment:
```suggestion
			$precision = (int) strpos(strrev(strval($number)), '.');
```

^ This is what you replaced `max(0, strlen(rtrim(sprintf('%.53F', abs($number + 0)), '0')) - strlen(strval(intval($number + 0))) - 1);` with in the other two cases, so I would think it is what we need in this case as well. Or am I missing something here?

Reply:
Probably. I was unsure on this, so wanted your input.